### ISSUE TITLE
[FileUploader] - Make the spinner optional when an error message is shown

### ISSUE THREAD

**Issue by guymaor86** [uber/baseweb#3356]
When an error message is shown in the File Uploader component, also the spinner is shown by default.
I think having the ability to not show the spinner by default would be great
![image](https://user-images.githubusercontent.com/8980910/82582328-2002d680-9b4f-11ea-8fb9-62726fee7e9a.png)

(thought it was a feature more than a bug but correct me if I'm wrong)
Also, I can try to do this with a little bit of help and pointers on where to start

**Comment by sandgraham** [comment]
I agree that the spinner looks a bit odd in the error state. The only thing you can do to avoid this at the moment is to set `progressAmount={0}` which will show an empty bar instead of the spinner. That isn't great so we should update component.

One option is to add a new prop, `{boolean} [hideProgressOnError = false]`. I believe there are still some cases where you might want to show the progress bar (not the progress spinner) during an error state, which is why I don't advocate for outright removing the progress section on error.

Another thing is we could update the design a bit so that the spinner stops spinning, has a red fill, etc. It's nice to have a visual cue something changed. Not totally sure what that should look like though.

**Comment by guymaor86** [comment]
thanks @sandgraham for the response
i tried already the `progressAmount={0}` but i liked the spinner better than the empty bar

i agree with you that we should not remove it the progress bar completely but have it as an option to remove it.

with that said, i'll try implementing the `{boolean} [hideProgressOnError = false]` you suggested

**Comment by sandgraham** [comment]
@guymaor86 - I spoke with our designers and it seems like we can just hide the spinner when there is an error. So there is no need for a new prop. If you were planning to do this soon feel free to open a PR and we will get it merged. Otherwise, we can likely get to it in the next couple weeks.

**Comment by guymaor86** [comment]
ok, i'll try to get to it soon

**Comment by guymaor86** [comment]
Hey @sandgraham 
I'm trying to push upstream but I'm getting permission denied.
How do I get added as a contributor?

**Comment by houmark** [comment]
@guymaor86 You fork Baseweb, create a branch in that fork and then a PR from that branch against the main Baseweb fork. If accepted, it will be merged in to the master of Baseweb.

**Comment by guymaor86** [comment]
oh ok, maybe worth adding that documentation to the contributing page? unless i missed it

**Comment by houmark** [comment]
Yeah, it's kinda mentioned that like most other Open Source projects, Baseweb expects PR's, but I agree that the contributor guide could have a step by step guide on how to contribute more specifically.

**Comment by guymaor86** [comment]
@houmark i can open a pr to add some documentation around that if you guys would like it
